src/choose-location/choose-location.resource.ts

```diff
@@ -0,0 +1,18 @@
+import { openmrsFetch, openmrsObservableFetch } from "@openmrs/esm-api";
+import { map } from "rxjs/operators";
+
+export function getLoginLocations(): Observable<object> {
+  return openmrsObservableFetch(
+    `/ws/rest/v1/location?tag=Login%20Location&v=custom:(uuid,display)`
+  ).pipe(map(({ data }) => data["results"]));
+}
+
+export function setSessionLocation(locationUuid: string): Promise<any> {
+  return openmrsFetch("/ws/rest/v1/appui/session", {
```

Comment:
I don't know if it's a good idea to be using a refapp endpoint here. I don't think we should have the refapp as a required dependency for an MF. I don't know another way of doing this - can you think of one?

Reply:
@mogoodrich suggested that this endpoint should be brought into REST if it's not a pain to do so. That's probably the right move. Is this a blocker?

Reply:
https://issues.openmrs.org/browse/RESTWS-758

Reply:
Does this work on openmrs-spa.org? I think that has refapp installed, right? I agree it would be good not to rely on more and more backend modules and the refapp is a big thing. However, if this is currently the only way, then let's do it.

We do need to add new backend dependencies to this project so that they show up correctly in the openmrs dev tools. See https://github.com/openmrs/openmrs-esm-home/blob/master/src/openmrs-backend-dependencies.ts for an example. Also check out the dev tools "backend dependencies" tab for more info.

Reply:
Yes, it does work on openmrs-spa.org.

I'll add the backend dependency.

Reply:
@jonathandick Are you cool with us merging this and planning to migrate the endpoint to REST after?

Reply:
@jonathandick ping?

Reply:
Weird, I never got this ping.

I suppose ok to merge as long as it won't break on AMPATH's implementation which does not have the ref app installed.

Reply:
If AMPATH does not have the App UI Module, this login with location flow will break on AMPATH's implementation. Is AMPATH using this login page?

Reply:
@jonathandick I've added a configuration parameter `chooseLocation: { enabled: true }`. Set this to `false` for AMPATH until we switch to the REST endpoint for setting session location.